torchtext/csrc/gpt2_bpe_tokenizer.cpp

```diff
@@ -61,28 +63,122 @@ std::vector<std::string> gpt2_bpe_pre_tokenizer(std::string input) {
   // - ELSE, add token to return list
   std::string token;
   std::vector<std::string> tokens;
-  re2::StringPiece inp(input);
   bool prepend_space = false;
-  while (kGPT2Regex.FindAndConsume(&inp, &token)) {
-    if (is_whitespace(token)) {
-      prepend_space = false;
-      if (inp.empty()) { // token is last token
-        tokens.push_back(token);
-      } else {
-        if (token.length() > 1) {
-          tokens.push_back(token.substr(0, token.length() - 1));
+  std::vector<std::string> index_matches;
+
+  /* Notes on handling Special Tokens:
```

Comment:
This thread is also very helpful for understanding spacing in BPE tokenization: https://discuss.huggingface.co/t/bpe-tokenizers-and-spaces-before-words/475